sct-commons/src/main/java/org/lfenergy/compas/sct/commons/DataTypeTemplatesService.java

```diff
@@ -33,14 +42,88 @@ public class DataTypeTemplatesService {
      */
     public boolean isDoModAndDaStValExist(TDataTypeTemplates dtt, String lNodeTypeId) {
         return lnodeTypeService.findLnodeType(dtt, lNodeType -> lNodeTypeId.equals(lNodeType.getId()))
-                .map(lNodeType -> doService.findDo(lNodeType, tdo -> MOD_DO_NAME.equals(tdo.getName()))
-                        .map(tdo -> doTypeService.findDoType(dtt, doType -> tdo.getType().equals(doType.getId()))
-                                .map(doType -> doType.getSDOOrDA().stream()
-                                                .filter(sdoOrDa -> sdoOrDa.getClass().equals(TDA.class))
-                                                .map(TDA.class::cast)
-                                                .anyMatch(tda -> STVAL_DA_NAME.equals(tda.getName())))
-                                 .orElse(false))
-                         .orElse(false))
-                 .orElse(false);
+                .flatMap(lNodeType -> doService.findDo(lNodeType, tdo -> MOD_DO_NAME.equals(tdo.getName()))
+                        .flatMap(tdo -> doTypeService.findDoType(dtt, doType -> tdo.getType().equals(doType.getId()))
+                                .map(doType -> sdoOrDAService.findSDOOrDA(doType, TDA.class, tda -> STVAL_DA_NAME.equals(tda.getName()))
+                                        .isPresent()))
+                ).orElse(false);
+    }
+
+    @Override
+    public Stream<DataAttributeRef> getAllDOAndDA(TDataTypeTemplates dtt) {
+        return dtt.getLNodeType().stream()
+                .flatMap(tlNodeType -> lnodeTypeService.getAllDOAndDA(dtt, tlNodeType, new DataAttributeRef()));
+    }
+
+    @Override
+    public Stream<DataAttributeRef> getFilteredDOAndDA(TDataTypeTemplates dtt, TAnyLN anyLn, DataAttributeRef filter) {
+        return lnodeTypeService.findLnodeType(dtt, tlNodeType -> tlNodeType.getId().equals(anyLn.getLnType()))
+                .stream()
+                .flatMap(tlNodeType -> lnodeTypeService.getFilteredDOAndDA(dtt, filter));
     }
+
+    @Override
+    public Optional<DataAttributeRef> findDOAndDA(TDataTypeTemplates dtt, String lNodeTypeId, DataAttributeRef dataAttributeRef) {
```

Comment:
For the same reason as earlier remarks.

This function only used in tests, remember that this PR tries to reduce certain difficulties.
For example, today if we want to search for a given DO(SDO..) and DA(BDA..) we ask ` AbstractLNAdapter#getDAI` then we verify whether the elements are included in the list or not. This function not search all, it allows us to search only given objects.

It can be very helpful to handle SCL object without adapters

Reply:
`DataAttributeRef` is being replaced with another object `DoLinkedToDa`